Add optional dependencies for the turbo package

### Is your enhancement related to a problem? Please describe.

The turbo package relies on the user architecture and platform to install the correct version. However, the optional dependencies need to be listed in the `package.json` file so turbo can pick the appropriate one.
E.g.
```json
"optionalDependencies": {
  "turbo-darwin-64": "1.9.3",
  "turbo-darwin-arm64": "1.9.3",
  "turbo-linux-64": "1.9.3",
  "turbo-linux-arm64": "1.9.3",
  "turbo-windows-64": "1.9.3",
  "turbo-windows-arm64": "1.9.3"
}
```

My setup:
- **Architecture**: Quad-Core Intel Core i7 (x64)
- **Platform**: macOS (Darwin)
- **node**: v18.18.0
- **npm**: v9.8.1

After running `npm install` this is the output:

```bash
npm ERR! [turbo] Failed to find package "turbo-darwin-64" on the file system
npm ERR! 
npm ERR! This can happen if you use the "--no-optional" flag. The "optionalDependencies"
npm ERR! package.json feature is used by turbo to install the correct binary executable
npm ERR! for your current platform. This install script will now attempt to work around
npm ERR! this. If that fails, you need to remove the "--no-optional" flag to use turbo.
npm ERR! 
npm ERR! node:internal/modules/cjs/loader:1077
npm ERR!   const err = new Error(message);
npm ERR!               ^
npm ERR! 
npm ERR! Error: Cannot find module 'turbo'
```

#### Links related to this issue
- https://github.com/vercel/turbo/issues/3328
- https://github.com/pnpm/pnpm/issues/5965
- https://sharp.pixelplumbing.com/install#cross-platform

### Describe alternatives you've considered

To fix it, I installed all turbo options with the `--save-optional` flag:
```
npm i turbo-darwin-64@1.9.3 turbo-darwin-arm64@1.9.3 turbo-linux-64@1.9.3 turbo-linux-arm64@1.9.3 turbo-windows-64@1.9.3 turbo-windows-arm64@1.9.3 --save-optional
```

This command updates the `package.json` and the `package-lock.json` files with all turbo alternatives.

After that change, running `npm install` (or `npm ci`) installs all packages without errors.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/package.json
+++ b/package.json
@@ -61,5 +61,13 @@
   },
   "dependencies": {
     "express": "^4.18.1"
+  },
+  "optionalDependencies": {
+    "turbo-darwin-64": "^1.9.3",
+    "turbo-darwin-arm64": "^1.9.3",
+    "turbo-linux-64": "^1.9.3",
+    "turbo-linux-arm64": "^1.9.3",
+    "turbo-windows-64": "^1.9.3",
+    "turbo-windows-arm64": "^1.9.3"
   }
 }

